File: Cpp/1603_Design_Parking_System.cpp
class ParkingSystem {
private:
    int park[3];
public:
    ParkingSystem(int big, int medium, int small) {
        park[0] = big;
        park[1] = medium;
        park[2] = small;
    }
    
    bool addCar(int carType) {
        if(park[carType-1] == 0){
            return false;
        }
        else{
            park[carType-1]--;
            return true;
        }
        //return park[carType-1]--;
    }
};

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * ParkingSystem* obj = new ParkingSystem(big, medium, small);
 * bool param_1 = obj->addCar(carType);
 */

File: Python3/1603_Design_Parking_System.py
class ParkingSystem:

    def __init__(self, big: int, medium: int, small: int):
        self.park = [big, medium, small]
        """
        self.park = []
        self.park.append(big)
        self.park.append(medium)
        self.park.append(small)
        """
        
    def addCar(self, carType: int) -> bool:
        self.park[carType-1] -= 1
        return self.park[carType-1] >= 0

# Your ParkingSystem object will be instantiated and called as such:
# obj = ParkingSystem(big, medium, small)
# param_1 = obj.addCar(carType)
